packages/components/src/components/pin-code/pin-code-utils.ts

```diff
@@ -0,0 +1,104 @@
+import type { FormState } from '../../utils/form/form-state';
+import type { Styles } from 'jss';
+import { consoleWarn, getTagNameWithoutPrefix, setAttributes } from '../../utils';
+
+export const PIN_CODE_TYPES = ['number', 'password'] as const;
+export type PinCodeType = (typeof PIN_CODE_TYPES)[number];
+
+export const PIN_CODE_LENGTHS = [4, 6] as const;
+export type PinCodeLength = (typeof PIN_CODE_LENGTHS)[number];
+
+export type PinCodeUpdateEvent = { value: string; isComplete: boolean };
+
+export type PinCodeState = FormState;
+
+export const hiddenInputSlotName = 'hidden-input';
+export const currentInputId = 'current-input';
+export const labelId = 'label';
+export const descriptionId = 'description';
+export const stateMessageId = 'state-message';
+
+export const removeSlottedSelector = (styles: Styles): Styles =>
+  Object.fromEntries(
+    Object.entries(styles).map(([key, value]) => {
+      value = typeof value === 'object' ? removeSlottedSelector(value as Styles) : value;
+      return [key.replace(/::slotted\(([^,]+)\)/g, '$1'), value];
+    })
+  );
+
+export const removeStyles = (selector: string, styles: Styles): Styles =>
+  Object.fromEntries(
+    Object.entries(styles)
+      .filter(([key]) => key !== selector)
+      .map(([key, value]) => {
+        value = typeof value === 'object' ? removeStyles(selector, value as Styles) : value;
+        return [key, value];
+      })
+  );
+
+export const warnIfInitialValueIsTransformed = (host: HTMLElement, length?: number): void => {
+  const warningPrefix = `Property value of component ${getTagNameWithoutPrefix(host)}:`;
+  consoleWarn(
+    warningPrefix,
+    length
+      ? `Provided value has too many characters and was truncated to the max length of ${length}.`
+      : 'Provided value contains characters that are not of type number, the value was therefore reset.'
+  );
+};
+
+export const isInputSingleDigit = (input: string): boolean => /^\d$/.test(input);
+
+export const hasInputOnlyDigitsOrWhitespaces = (input: string): boolean => /^[\d ]+$/.test(input);
+
+export const getInputValue = (pinCodeElements: HTMLInputElement[]): string =>
```

Comment:
Name suggests there is only a single input but it is an array.
Perhaps something like `getConcatenatedInputValues` or `getConcatenatedInputsValue` would fit better.